### ISSUE TITLE
Parent/Child document support

### ISSUE THREAD

**Issue by jpeyper** [searchkit/searchkit#241]
Following the example given by ElasticSearch (https://www.elastic.co/guide/en/elasticsearch/guide/current/indexing-parent-child.html), we have a similar requirement to store documents like this.

We have two documents types:
Account - 1 for every account
Transaction - 100s for each account parent

We want to be able to search for accounts with certain transaction types, we do not want to store this as one document in Elastic. 

Ideally, this would look similar to the refinement list filter, but other support for ranges would be great!

**Comment by ssetem** [comment]
@jpeyper do you mind pasting a snippet of both your account and transaction mapping config, so we can be sure we are coding this correctly, thanks!

**Comment by jpeyper** [comment]
We haven't actually setup the mapping yet. This is just how we were
thinking about doing it.

I'll knock something up on Monday (Aussie time) when I'm back at work and
post it.

In the meantime, if there is anything we should avoid that will make things
a lot harder for you, let me know.

**Comment by jpeyper** [comment]
Sorry for taking so long... apparently Aussie time means being slack and forgetful... so not completely inaccurate :P

So this is what we are thinking (this isn't our actual mapping, just a cut down version of what we are wanting).

Using ElasticSearch 2.3
```
POST /example
{
  "mappings": {
    "account": {
      "properties": {
        "accountId": {
          "type": "string",
          "index": "not_analyzed"
        },
        "data": {
          "properties": {
            "accountType": {
              "type": "string"
            }
          }
        }
      }
    },
    "transaction": {
      "_parent": {
        "type": "account"
      },
      "properties": {
        "transactionId": {
          "type": "string",
          "index": "not_analyzed"
        },
        "data": {
          "properties": {
            "transactionType": {
              "type": "string"
            },
            "effectiveDate": {
              "type": "date",
              "format": "yyyy-MM-dd"
            }
          }
        }
      }
    }
  }
}
```
```
POST /example/account/1
{
    "accountId": "1",
    "data": {
        "accountType": "Active"
    }
}
```
```
POST /example/account/2
{
    "accountId": "2",
    "data": {
        "accountType": "Inactive"
    }
}
```
```
POST /example/transaction/1?parent=1
{
    "transactionId": "1",
    "data": {
        "transactionType": "Buy",
        "effectiveDate": "2016-04-01"
    }
}
```
```
POST /example/transaction/2?parent=1
{
    "transactionId": "2",
    "data": {
        "transactionType": "Sell",
        "effectiveDate": "2016-04-02"
    }
}
```
```
POST /example/transaction/3?parent=2
{
    "transactionId": "3",
    "data": {
        "transactionType": "Buy",
        "effectiveDate": "2016-04-03"
    }
}
```
```
POST /example/transaction/4?parent=2
{
    "transactionId": "4",
    "data": {
        "transactionType": "Buy",
        "effectiveDate": "2016-04-04"
    }
}
```

From here we would like to be able to filter members that have particular transaction types, or transactions during a particular range. Showing counts on the refinement lists for the child aggregations would be nice, however my testing seems to show that it would display the counts of the child documents, not the parent.

For example:
```
POST /example/_search
{
  "size": 0,
  "aggs": {
    "accountTypes": {
      "terms": { 
        "field": "data.accountType"
      }
    },
    "transactions": {
      "children": { 
        "type": "transaction"
      },
      "aggs": {
        "transactionTypes": {
          "terms": { 
            "field": "data.transactionType"
          }
        }
      }
    }
  }
}
```
returns (there are only 2 accounts and 4 transactions)
```
{
  "took": 7,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "failed": 0
  },
  "hits": {
    "total": 6,
    "max_score": 0,
    "hits": []
  },
  "aggregations": {
    "accountTypes": {
      "doc_count_error_upper_bound": 0,
      "sum_other_doc_count": 0,
      "buckets": [
        {
          "key": "active",
          "doc_count": 1
        },
        {
          "key": "inactive",
          "doc_count": 1
        }
      ]
    },
    "transactions": {
      "doc_count": 4,
      "transactionTypes": {
        "doc_count_error_upper_bound": 0,
        "sum_other_doc_count": 0,
        "buckets": [
          {
            "key": "buy",
            "doc_count": 3
          },
          {
            "key": "sell",
            "doc_count": 1
          }
        ]
      }
    }
  }
}
```
I'm not sure if this is solvable with my current elastic search knowledge.

A lot of stuff there so happy to clarify anything and we are definitely open for suggestion of a better way of doing things.

**Comment by ssetem** [comment]
thanks @jpeyper

**Comment by Rhubbikull** [comment]
Hi, 
Just wondering if there been any development on this improvement?

**Comment by ssetem** [comment]
@Rhubbikull currently in progress. Hopefully done soon

**Comment by ssetem** [comment]
@Rhubbikull @jpeyper 

We have done a prerelease 0.10.0-4 with nested + parent/child support

[FieldOptions Docs](https://github.com/searchkit/searchkit/blob/develop/docs/core/FieldOptions.md)

Would you mind testing this? Thanks!

**Comment by jpeyper** [comment]
Sorry for ignoring this for so long... been very busy.

We have moved away from using searchkit in favour of our own frontend implementation.

The decision was made that we did not want to open our elasticsearch cluster up to the internet so we had to put an [BFF layer](http://samnewman.io/patterns/architectural/bff/) in-front of elastic, allowing us to lock down exactly what our clients could search for and implementing our security model.

So we have looked at this at all, nor do we have plans to. Thanks for all your time and effort, searchkit and its developers are really great.
